run.py

```diff
@@ -239,12 +252,168 @@ def test():
     print("All tests for mtriage done.")
 
 
+def viewer(args):
+
+    """ Must be invoked with an input folder and viewer-plugin e.g.:
+
+            python run.py viewer -i <derived_folder> -v <viewer_plugin>
+
+        Server requires that input folder contains directories corresponsing to elements (where directory names
+        are element ids.) Each element directory must contain a json file containing:
+
+        {
+            "id": <element_id>,
+            "media": [
+                <media1.mp3>
+                <media1.mp4>,
+                <media1.png>,
+                etc.
+            ]
+        }
+
+        As well as any other data to be consumed by the viewr-plugin. Media are any other files
+        in the element folder that need to be available to the viewer-plugin.
+
+        Viewers must be yarn buildable apps living in the src/lib/viewers folder. Their name is taken to be the name
+        of their outer directory.
+
+        Once launched the viewer plugin is available at http://localhost:8081/
+
+        The server is available at http://localhost:8080/, with available endpoints:
+
+        elements                                    - returns list of element ids
+        element?id=<element_id>                     - serves the element's json file
+        element?if=<element_id>&media=<media_file>  - serves the media file associated with element
+
+    """
+
+    folder = args.input
+    viewer = args.viewer
+
+    if args.input == None:
+        raise InvalidArgumentsError("No input directory supplied for viewer plugin.")
+
+    if args.viewer == None:
+        raise InvalidArgumentsError("No viewer plugin name supplied.")
+
+    if not os.path.exists(folder):
+        raise WorkingDirectorNotFoundError(folder)
+
+    shutil.rmtree("src/server/elements/")
+    os.makedirs("src/server/elements/")
+
+    element_folders = [f for f in os.listdir(folder) if os.path.isdir(folder + "/" + f)]
+
+    for e in element_folders:
+        f = str(folder) + "/" + str(e)
+        for file in os.listdir(f):
+            if not os.path.exists("src/server/elements/" + e):
+                os.makedirs("src/server/elements/" + e)
+            os.symlink(
+                "/mtriage/" + f + "/" + file, "src/server/elements/" + e + "/" + file
+            )
+
+    print("Creating container to build server...")
+    print("----------------------------------")
+    try:
+        sp.call(
+            [
+                "docker",
+                "build",
+                "-t",
+                "{}:dev".format(SERVER_NAME),
+                "-f",
+                "src/server/server.Dockerfile",
+                "src/server",
+            ]
+        )
+        print("Build successful, attempting to run")
+    except:
```

Comment:
We should probably check whether an image for the particular viewer already exists in the local image repo, to save having to rebuild it. There could also be a `--force` option that always rebuilds the image from scratch.